server_admin/topics/vault.adoc

```diff
@@ -54,6 +62,153 @@ Here is the equivalent configuration using CLI commands:
 /subsystem=keycloak-server/spi=vault/provider=files-plaintext/:add(enabled=true,properties={dir => "${jboss.home.dir}/standalone/configuration/vault"})
 ----
 
-NOTE: There is at most one vault provider active per {project_name} instance
-at any given time, and the vault provider in each instance within the cluster
-have to be configured consistently.
+=== Elytron Credential Store Vault Provider
+
+{project_name} also provides support for reading secrets stored in an Elytron credential store. The `elytron-cs-keystore`
+vault provider is capable of retrieving secrets from the keystore-based implementation of the credential store, which
+is also the default implementation provided by Elytron.
+
+This credential store is backed by a keystore (`JCEKS` is the default format, but it is possible to use other formats such as `PKCS12`)
+and users can create and manage the store contents using either the `elytron` subsystem in WildFly/JBoss EAP, or using the
+`elytron-tool.sh` script.
+
+To use this provider, you have to declare the `elytron-cs-keystore` in the `keycloak-server` subsystem and set the location
+and master secret of the keystore that was created by Elytron. An example of the minimal configuration for the provider is shown bellow:
+
+[source, xml]
+----
+<spi name="vault">
+    <default-provider>elytron-cs-keystore</default-provider>
+    <provider name="elytron-cs-keystore" enabled="true">
+        <properties>
+            <property name="location" value="${jboss.home.dir}/standalone/configuration/vault/credential-store.jceks" />
+            <property name="secret" value="secretpw1!"/>
+        </properties>
+    </provider>
+</spi>
+----
+
+If the underlying keystore has a format other than `JCEKS`, this has to be informed using the `keyStoreType`:
```

Comment:
"this" should always precede a noun to make it clear what "this" is referring to.